### ISSUE TITLE
CreateOauthTables migration buggy

### ISSUE THREAD

**Issue by christoph-buente** [pelle/oauth-plugin#6]
When running ./script/generate oauth_provider a migration for nessecary tables is generated. The indexes that are created should be uniqe
add_index :client_applications, :key, :unique

Correct Syntax is:
add_index :client_applications, :key, :unique => true

**Comment by brugidou** [comment]
I agree with you, but strangely enough it didn't throwe an error after a db:migrate... And my indexes are unique, i guess that the ":unique" syntax works too.

**Comment by christoph-buente** [comment]
It didn't work for me, that's why i had to modify it. After i had it running, this info might be usefull for others.

**Comment by brugidou** [comment]
OK, i guess this should be fixed then but i don't know who if pelle is hearing us...

(for info, i'm with rails 2.3.5 and ruby 1.9.1)

**Comment by cantlin** [comment]
Can anyone shed any light on this response to `rake db:migrate` after `rails g oauth_consumer`?

<code>Mysql2::Error: Specified key was too long; max key length is 767 bytes: CREATE UNIQUE INDEX `index_consumer_tokens_on_token` ON `consumer_tokens` (`token`)</code>

**Comment by christoph-buente** [comment]
This is related to the mysql charset. change it to UTF8, that should do the trick

Christoph

Am 12.03.2011 um 19:21 schrieb cantlinreply@reply.github.com:

> Can anyone shed any light on this response to rake db:migrate?
> 
> Mysql2::Error: Specified key was too long; max key length is 767 bytes: CREATE UNIQUE INDEX `index_consumer_tokens_on_token` ON `consumer_tokens` (`token`)
> 
> https://github.com/pelle/oauth-plugin/issues/6#comment_864063

**Comment by cantlin** [comment]
Thanks Christoph, that did it.

**Comment by wleeperogs** [comment]
I had the unique issue as well.  Running Rails 2.3.8 and Ruby 1.8.7.  The other index created in this migration has the => true in it.

**Comment by bournenapster** [comment]
I edited mysql for utf8 but it did not worked any idea ?

 vi /etc/mysql/my.cnf

There are several sections in the file. Modify two of them – “client” and “mysqld” as shown below to configure mySql for UTF-8 support:

...
[client]

default-character-set=utf8

...
[mysqld]

default-character-set=utf8
character-set-server=utf8
default-collation=utf8_unicode_ci

any suggestions i am looking for a quick solution ?

**Comment by christoph-buente** [comment]
If the table was created with latin1 you have to alter it first. Setting the default-character set doesn't affect existing tables. 

alter table TABLE_NAME change COLUMN_NAME COLUMN_NAME COLUMN_DATA_TYPE character set utf8;
